p2p/proto/receipt.proto

```diff
@@ -42,6 +44,7 @@ message Receipt {
     repeated           MessageToL1 messages_sent = 3;
     ExecutionResources execution_resources       = 4;
     string             revert_reason             = 5;
+    Events             events                    = 6;
```

Comment:
These were intentionally separated from receipts. The idea was there are use cases where events are needed, but not the execution resources.

Reply:
With the current design, it is not possible to serve `starknet_getTransactionReceipt` because we don't have any mapping of which events belong to which transaction receipt, we are only aware of when events belong to a block. The above change allows us to do that with out internal representation.

Reply:
The mapping is the tx hash. And in the long run we'll probably change the RPC to match at some point.

Reply:
I don't think there's a way in the current P2P representation to associate a tx hash with an event, only with blocks as @IronGauntlets mentioned above.

Reply:
Correct. So there should be a collection of events a tx outputted with its hash

Reply:
Why not just modify the events reply to include an event sequence to transaction hash mapping [as I suggested in the first PR](https://github.com/starknet-io/starknet-p2p-specs/pull/8#discussion_r1371276710)

Reply:
We're talking of the same thing. TxnEvents is a collection of events a tx outputted with its hash.

Reply:
And yeah, it should have been added in the first PR. Better late than never.

Reply:
@CHr15F0x I will add your suggestion.

Reply:
Please view this [PR](https://github.com/starknet-io/starknet-p2p-specs/pull/18)